edk2toolext/environment/version_aggregator.py

```diff
@@ -29,20 +29,22 @@ def ReportVersion(self, key, value, versionType):
         versionType -- The method of categorizing what is being reported. See VersionTypes for details.
         """
         if key in self.Versions:
-            if self.Versions[key]["version"] == value:
-                self._logger.warning("version_aggregator: This {0}:{1} key/value pair "
-                                     "was already registered".format(key, value))
+            old_version = self.Versions[key]
+            if old_version["version"] == value and old_version["path"] == path:
+                pass
             else:
                 error = "version_aggregator: {0} key registered with a different value\n\t" \
-                        "Old:{1}\n\tNew:{2}".format(key, self.Versions[key]["version"], value)
+                        "Old:{1}@{3}\n\tNew:{2}@{4}\n".format(
+                            key, old_version["version"], value, old_version["path"], path)
```

Comment:
Is this fine to print if "path" is left as None? Was the None case verified?

Reply:
Confirmed that None case works fine

Reply:
ValueError: Intel.ICL.GOP key registered with a different value
        Old:14.0.1027.1@None
        New:14.0.102.1@None